server/subscribe.go

```diff
@@ -360,22 +365,95 @@ func (p *Plugin) listChannelSubscriptions() (string, error) {
 		return "", err
 	}
 
+	teamDMGMSubs, err := p.getSubscriptionsOrderedByTeamDMandGM()
+	if err != nil {
+		return "", err
+	}
+
 	rows := []string{}
+	printDMGMHeader := true
+
+	for _, teamSubs := range teamDMGMSubs {
+
+		// create team header for channels.  Only print header for DMs and GMs channels once.
+		if teamSubs.TeamId != "" {
+			channelTeam, appErr := p.API.GetTeam(teamSubs.TeamId)
+			if appErr != nil {
+				return "", appErr
+			}
+			rows = append(rows, fmt.Sprintf("\n### %s", channelTeam.DisplayName))
+		} else {
+			if printDMGMHeader {
+				rows = append(rows, fmt.Sprintf("\n### %s", "Group and Direct Messages"))
+				printDMGMHeader = false
+			}
+		}
+
+		var printChannelName = true
+		for _, subId := range teamSubs.SubIds {
+			sub := subs.Channel.ById[subId]
+
+			channel, appErr := p.API.GetChannel(sub.ChannelId)
+			if appErr != nil {
+				return "", errors.New("Failed to get channel")
+			}
+
+			// only print channel name once for all subscriptions
+			if printChannelName {
+				rows = append(rows, fmt.Sprintf("* **~%s** (%d):", channel.Name, len(teamSubs.SubIds)))
+				printChannelName = false
```

Comment:
I don't think this loop separates the sub ids between different channels. Wouldn't this code only print out the channel for the first subscription? `printChannelName` is never set to `true` except for just before this loop begins. And this loop is for all subscription ids for a given team, not grouped by channel.

I see that you only have one channel with subscriptions per team in your screenshot. Do you mind writing a unit test with two channels in the same team?

Reply:
Confirmed this issue occurs.  I'm reworking the code to account for multiple channels with subscriptions in the same team.

Reply:
Did you include the fix for this? Could you write a unit test for this case?

Reply:
Will re-review once this is addressed @jfrerich

Reply:
@mickmister, Tests added in last two commits